### ISSUE TITLE
literal_only_boolean_expressions should handle ??

### ISSUE THREAD

**Issue by apwilson** [dart-lang/linter#517]
This triggers the literal_only_boolean_expressions rule but shouldn't imo.
```
if (foo?.bar ?? false) {
}
```

**Comment by srawlins** [comment]
`literal_only_boolean_expressions` no longer reports on such an expression.

```
$ dartanalyzer --version
dartanalyzer version 2.0.0-dev.63.0
$ cat c.dart 
void main() {
  int b;
  if (b?.isEven ?? true) {
    print('nay');
  }
}
$ dartanalyzer c.dart 
Analyzing c.dart...
No issues found!
```
